File: app/src/main/java/pl/pzienowicz/zditmszczecinlive/model/Info.java
package pl.pzienowicz.zditmszczecinlive.model;

import com.google.gson.annotations.SerializedName;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;

public class Info {

    private DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");

    @SerializedName("od")
    private String from;

    @SerializedName("do")
    private String to;

    @SerializedName("opis")
    private String description;

    public String getTo() {
        return to;
    }

    public String getFrom() {
        return from;
    }

    public String getDescription() {
        return description;
    }

    public String getToDate() {
        if(getTo() == null) {
            return null;
        }

        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(Long.parseLong(getTo()) * 1000);

        return dateFormat.format(calendar.getTime());
    }

    public String getFromDate() {
        if(getFrom() == null) {
            return null;
        }

        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(Long.parseLong(getFrom()) * 1000);

        return dateFormat.format(calendar.getTime());
    }
}

File: app/src/main/java/pl/pzienowicz/zditmszczecinlive/model/Info.kt
package pl.pzienowicz.zditmszczecinlive.model

import java.text.SimpleDateFormat

data class Info(
    val valid_from: String,
    val valid_to: String? = null,
    val description: Description
) {
    val toDate: String?
        get() {
            if (valid_to == null) {
                return null
            }

            val dateFormatFrom = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS")
            val dateFormatTo = SimpleDateFormat("yyyy-MM-dd")

            val date = dateFormatFrom.parse(valid_to)
            return date?.let {
                dateFormatTo.format(it)
            }
        }

    val fromDate: String
        get() {
            val dateFormatFrom = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS")
            val dateFormatTo = SimpleDateFormat("yyyy-MM-dd")

            val date = dateFormatFrom.parse(valid_from)
            return date?.let {
                dateFormatTo.format(it)
            } ?: ""
        }

    data class Description(val pl: String)
}
